File: backend/internal/models/game.go
package models

import "time"

type Status string

const (
	SCHEDULED   Status = "Scheduled"
	IN_PROGRESS Status = "In Progress"
	FINAL       Status = "Final"
)

type Game struct {
	DbModel
	SeasonID uint      `json:"season_id" validate:"required"`
	Start    time.Time `json:"start" validate:"required"`
	Venue    *Venue     `json:"venue"`
	VenueID  *uint      `json:"venue_id" validate:"required"`
	Status   Status    `json:"status" validate:"required"`

	HomeTeam            *Team   `json:"home_team"`
	HomeTeamID          uint   `json:"home_team_id" validate:"required"`
	HomeTeamRoster      *Roster `json:"home_team_roster"`
	HomeTeamRosterID    *uint   `json:"home_team_roster_id" validate:"required"`
	HomeTeamLockerRoom  string `json:"home_team_locker_room" validate:"required"`
	HomeTeamShotsOnGoal int    `json:"home_team_shots_on_goal" validate:"required"`
	HomeTeamScore       int    `json:"home_team_score" validate:"required"`

	AwayTeam            *Team   `json:"away_team"`
	AwayTeamID          uint   `json:"away_team_id" validate:"required"`
	AwayTeamRoster      *Roster `json:"away_team_roster"`
	AwayTeamRosterID    *uint   `json:"away_team_roster_id" validate:"required"`
	AwayTeamLockerRoom  string `json:"away_team_locker_room" validate:"required"`
	AwayTeamShotsOnGoal int    `json:"away_team_shots_on_goal" validate:"required"`
	AwayTeamScore       int    `json:"away_team_score" validate:"required"`

	ScoreKeeper        *User `json:"score_keeper"`
	ScoreKeeperID      uint  `json:"score_keeper_id" validate:"required"`
	PrimaryReferee     *User `json:"primary_referee"`
	PrimaryRefereeID   *uint `json:"primary_referee_id" validate:"required"`
	SecondaryReferee   *User `json:"secondary_referee"`
	SecondaryRefereeID *uint `json:"secondary_referee_id" validate:"required"`
}

File: app/src/models/Game.ts
import { RosterId, SeasonId, TeamId, UserId, VenueId } from './ids'
import type { DbModel, Roster, Team, User, Venue } from './index'
import { TimeString } from './types/timeString'


export enum GameStatus {
  SCHEDULED = 'Scheduled',
  IN_PROGRESS = 'In Progress',
  FINAL = 'Final'
}

export interface Game extends DbModel {
  season_id: SeasonId
  start: TimeString
  venue: Venue
  venue_id: VenueId
  status: GameStatus
  
  // TODO: This might change to home: { ... } and away: { ... }
  home_team: Team
  home_team_id: TeamId
  home_team_roster: Roster
  home_team_roster_id: RosterId
  home_team_locker_room: string
  home_team_shots_on_goal: number
  home_team_score: number
  
  away_team: Team
  away_team_id: TeamId
  away_team_roster: Roster
  away_team_roster_id: RosterId
  away_team_locker_room: string
  away_team_shots_on_goal: number
  away_team_score: number
  
  score_keeper: User
  score_keeper_id: UserId
  primary_referee: User
  primary_referee_id: UserId
  secondary_referee: User
  secondary_referee_id: UserId
}

export type { GameId } from './ids'
